### DISCUSSION TITLE
Issue in exporting vm metrics

### DISCUSSION THREAD

**Discussion by vijaypr1** [apache/skywalking#8938] [Q&A] [answered]
Hi Team,
I m new to Skywalking and currently exploring the same..
I m facing issue while exporting vm node metrics to sky walking for viewing... the screen is blank under VM section and none of the metrics are showing up
I have used skywalking 8.4 and has followed ::::: https://skywalking.apache.org/blog/2021-02-07-infrastructure-monitoring/
I have created the VM instance in GCP
Insalled and started node exporter -- ./node_exporter (the metrics are showing up correctly)
Downloaded Skywalking from distribution and started using startup.sh script under bin dir
All the node agent, skywalking OAP, otel exporter exists in the same vm..

As per the instruction i have created docker-compose.yaml,  otel-collector-config.yaml, as given in the instructions and while running docker compose up, it is running but below error is noticed:::

otel-collector_1  |     go.opentelemetry.io/collector/processor/batchprocessor/batch_processor.go:144
otel-collector_1  | 2022-04-23T11:45:41.047Z    error   exporterhelper/queued_retry.go:176      Exporting failed. Dropping data. Try enabling sending_queue to survive temporary failures.      {"kind": "exporter", "name": "opencensus", "dropped_items": 5}
otel-collector_1  | go.opentelemetry.io/collector/exporter/exporterhelper.(*queuedRetrySender).send
otel-collector_1  |     go.opentelemetry.io/collector/exporter/exporterhelper/queued_retry.go:176
otel-collector_1  | go.opentelemetry.io/collector/exporter/exporterhelper.NewMetricsExporter.func2
otel-collector_1  |     go.opentelemetry.io/collector/exporter/exporterhelper/metrics.go:97
otel-collector_1  | go.opentelemetry.io/collector/consumer/consumerhelper.ConsumeMetricsFunc.ConsumeMetrics
otel-collector_1  |     go.opentelemetry.io/collector/consumer/consumerhelper/metrics.go:29
otel-collector_1  | go.opentelemetry.io/collector/service/internal/fanoutconsumer.metricsConsumer.ConsumeMetrics
otel-collector_1  |     go.opentelemetry.io/collector/service/internal/fanoutconsumer/consumer.go:51
otel-collector_1  | go.opentelemetry.io/collector/processor/batchprocessor.(*batchMetrics).export
otel-collector_1  |     go.opentelemetry.io/collector/processor/batchprocessor/batch_processor.go:285
otel-collector_1  | go.opentelemetry.io/collector/processor/batchprocessor.(*batchProcessor).sendItems
otel-collector_1  |     go.opentelemetry.io/collector/processor/batchprocessor/batch_processor.go:183
otel-collector_1  | go.opentelemetry.io/collector/processor/batchprocessor.(*batchProcessor).startProcessingCycle
otel-collector_1  |     go.opentelemetry.io/collector/processor/batchprocessor/batch_processor.go:144
otel-collector_1  | 2022-04-23T11:45:41.047Z    warn    batchprocessor/batch_processor.go:184   Sender failed  {"kind": "processor", "name": "batch", "error": "EOF"}

![image](https://user-images.githubusercontent.com/104262050/164896632-b001751a-5652-4571-8674-9f21e389986c.png)

**Comment by wu-sheng** [comment, marked as answer]
Showcase is the best choice to give a 1st try, https://skywalking.apache.org/docs/#SkyWalkingShowcase You just need a k8s.

Also, you are describing OpenTelemetry issues, which is not SkyWalking's responsibility. Showcase includes which version we use, and how we set up the OpenTelemetry collector.

**Reply by chenxiaohu** [reply]
@wu-sheng 
I try showcase lastest and v8 with docker compose (Docker Desktop Mac, 4.9.1 (81317)), everything is ok but vm not work, not error, but no data into oap server( I check data in elasticsearch).

**Reply by wu-sheng** [reply]
We tried, and it works. The demo env on the website is deploying this showcase in our INFRA only

**Reply by chenxiaohu** [reply]
I think the demo env on the website is k8s deploy, not docker compse, is that?

**Reply by wu-sheng** [reply]
OK, got your point now. I am not sure about docker compose, we are rarely using it.

**Reply by chenxiaohu** [reply]
Thanks, I think docker compose is much simpler than k8s for development env, and skywalking e2e test also use compose.

**Reply by wu-sheng** [reply]
Because docker-compose has limited features, such as no service mesh and faas. Also, there are several more features can't be done on k8s, which is why we usually use k8s.
Compose is good for testing, yes, we are using it. But you are asking for a quick and easy demo setup.

Anyway, I just want to explain why we work more on k8s. Why does this not work on VM? I think you could follow this doc to do some diagnosis https://skywalking.apache.org/docs/main/latest/en/setup/backend/backend-vm-monitoring/#setup.

**Reply by chenxiaohu** [reply]
OK, Thanks for your quickly reply. I try it:)

**Reply by wankai123** [reply]
If you using OAP version < 9.1.0, the OTEL collector version should < 0.30.0,  because the OTEL collector protocol and config changed.
And check your version and config and read the comment 
 here:https://github.com/apache/skywalking/blob/61278ddd0ef22d7aba0e6834d765eb6027c07770/test/e2e-v2/cases/vm/prometheus-node-exporter/otel-collector-config.yaml#L38

I suggest you use  OTEL 0.29.0, if your OAP is 8.4.

**Reply by chenxiaohu** [reply]
@wu-sheng @wankai123 

https://github.com/apache/skywalking-showcase/blob/main/deploy/platform/docker/config/otel-collector-config.yaml

just change  job_name from "otel-collector" to "vm-monitoring", It work ：）

**Reply by wu-sheng** [reply]
Would you like to submit a pull request to fix on the Showcase repository?

**Reply by chenxiaohu** [reply]
OK ~ @wu-sheng

**Reply by wu-sheng** [reply]
https://github.com/apache/skywalking-showcase/pull/52
